content/guides/developer-guides/delfin/driver-developer-guide/_index.md

```diff
@@ -0,0 +1,241 @@
+---
+title: Driver Developer Guide
+menuTitle: Driver 
+description: "Delfin Driver Development Guide"
+weight: 10
+disableToc: false
+tags: ["developer guide", "delfin", "driver"]
+---
+
+
+
+Delfin is an Infrastructure Management framework under SODA foundation projects for Heterogeneous storage backends (cloud-enabled, offline-storage, etc.) for management, status collection, telemetry and alerting.
+
+## Goals
+
+  - Enable third parties to integrate a new backend to SODA Delfin framework
+  - Collect the steps needed to integrate a new backend to SODA Delfin framework
+  - Details the interfaces that new backend driver needs to be implemented
+
+## Motivation and background
+
+Delfin is an Infrastructure Management framework developed in Python programming language. It provides a Python plugin interface for adding third party drivers thereby supporting third party backends. The third party driver needs to implement the interfaces defined in the Driver python class, for the framework to use the third party driver. Once drivers implement the interfaces with the details of backend storage, delfin can manage the backend.
+
+## Non-Goals
+
+  - Support backend specific implementation details 
+  - Explain internal or higher level framework specific details
+
+## License
+
+Delfin project is Licensed under the *Apache License, Version 2.0*. For a third party driver to be merged to Delfin project repository, please ensure it uses compatible License.
+
+## Third party driver integration
+
+SODA Delfin project already contains some [drivers](https://github.com/sodafoundation/delfin/tree/master/delfin/drivers), which can be used as reference by the new third party driver developers.
+
+Existing Delfin Drivers for reference:
+  - fake_driver - This is a dummy/sample driver used for testing purpose
+  - Huawei - This driver implements Huawei's OeanStor backend
+  - VMAX - This driver implements Dell EMC's VMAX storage backend
+
+### Code changes needed
+
+* Add driver plugin 'entry points' to the file 'setup.py'.
+  ```python
+
+    'delfin.storage.drivers': [
+        ...
+        'sample_vendor sample_model = delfin.drivers.sample_vendor:SampleStorageDriver',
+        ...
+    ]
+
+  ```
+
+* Create driver source code folder under `<delfin path>/delfin/drivers/`
+  ```bash
+  mkdir -p /path-to-delfin/delfin/drivers/sample_vendor
+  touch /path-to-delfin/delfin/drivers/sample_vendor/__init__.py
+  ```
+
+* Extend base class StorageDriver defined in `<delfin path>/delfin/drivers/driver.py`, to implement a new driver.
+
+  ```python
```

Comment:
Can we point to the github file location, rather than pasting code?

Reply:
removed unwated code, git hub reference  will be too much to undestand. we want show  minimum required.